source/JsMaterialX/JsMaterialXFormat/post.js

```diff
@@ -0,0 +1,448 @@
+                }
+                path.replace(new RegExp(pathSep, "g"), "/");
+            } else if (ENVIRONMENT_IS_WEB) {
+                var link = document.createElement("a");
+                link.href = path;
+                if (link.origin + link.pathname + link.search + link.hash === path) {
+                    path = wasmRootFolder + link.pathname;
+                }
+            } else {
+                throw new Error("Unknown environment!");
+            }
+            pathList[i] = path;
+        }
+        if (typeof paths === 'string') {
+            return pathList.join(Module.PATH_LIST_SEPARATOR);
+        }
+    }
+
+    // Tweak the user-provided search path so that it works in the WASM FS.
+    function getWasmSearchPath(searchPath, wasmRootFolder) {
+        var wasmSearchPath = searchPath.split(PATH_LIST_SEPARATOR);
+        makeWasmAbsolute(wasmSearchPath, wasmRootFolder);
+        wasmSearchPath.push(getWasmCwd(wasmRootFolder));
+        wasmSearchPath = removeDuplicates(wasmSearchPath);
+        wasmSearchPath = wasmSearchPath.join(Module.PATH_LIST_SEPARATOR);
+        return wasmSearchPath;
+    }
+
+    // Parse a file to get includes.
+    function getIncludes(file) {
+        var includeRegex = /<xi:include href="(.*)"\s*\/>/g;
+        var matches = file.matchAll(includeRegex);
+        var includes = [];
+        for (var match of matches) {
+            includes.push(match[1]);
+        }
+        return includes;
+    }
+
+    // Load a file depending on the environment.
+    function loadFile(fileToLoad, searchPaths) {
+        var promise;
+        if (ENVIRONMENT_IS_WEB) {
+            promise = fetchXml(fileToLoad, searchPaths);
+        } else if (ENVIRONMENT_IS_NODE) {
+            promise = loadXml(fileToLoad, searchPaths);
+        } else {
+            throw new Error("Unknown environment!");
+        }
+        return promise;
+    }
+
+    // Track folders and files that have been created in the WASM file system, to delete them again later.
+    function trackPath(path, filesUploaded, isFile = false) {
+        if (isFile) {
+        // Remember full file path
+            if (!filesUploaded.files) {
+                filesUploaded.files = [];
+            }
+            filesUploaded.files.push(path);
+        } else {
+            // Remember paths in inverse order of creation, to delete them again later.
+            if (!filesUploaded.folders) {
+                filesUploaded.folders = [];
+            }
+            filesUploaded.folders.splice(0, 0, path);
+        }
+    }
+
+    // Store a file in the WASM file system. File is expected to be an absolute path, starting with wasmRootFolder.
+    function createInWasm(file, data, filesUploaded, wasmRootFolder, isFile = true) {
+        // Create folders if necessary.
+        var folders;
+        if (isFile) {
+           folders = file.substring(1, file.lastIndexOf("/")).split("/");
```

Comment:
While reading through the code, I'm a little confuse when to use pathSep, PATH_LIST_SEPARATOR and the string literal "/". Can we substitute the use of "/" by a variable so that it is more readable when to use it? Something like "browserSep" 🤷‍♀️

Reply:
I've added some comments on the variables to clarify their purpose.